test/jb_controller/distribute_reserved_token_of.test.js

```diff
@@ -0,0 +1,483 @@
+        await expect(tx)
+          .to.emit(jbController, 'DistributeToReservedTokenSplit')
+          .withArgs(
+            timestamp,
+            1,
+            PROJECT_ID,
+            [
+              split.preferClaimed,
+              split.percent,
+              split.lockedUntil,
+              split.beneficiary,
+              split.allocator,
+              split.projectId,
+            ],
+            RESERVED_AMOUNT / splits.length,
+            caller.address,
+          );
+      }),
+      await expect(tx)
+        .to.emit(jbController, 'DistributeReservedTokens')
+        .withArgs(
+          timestamp,
+          1,
+          PROJECT_ID,
+          projectOwner.address,
+          RESERVED_AMOUNT,
+          0,
+          MEMO,
+          caller.address,
+        ),
+    ]);
+  });
+
+  it(`Should send all left-over tokens to the project owner`, async function () {
+    const { addrs, projectOwner, jbController, mockTokenStore, mockSplitsStore, timestamp } =
+      await setup();
+    const caller = addrs[0];
+    const splitsBeneficiariesAddresses = [addrs[1], addrs[2]].map((signer) => signer.address);
+
+    const splits = makeSplits({
+      count: 2,
+      beneficiary: splitsBeneficiariesAddresses,
+      preferClaimed: true,
+      redemptionRate: 0,
+    });
+
+    splits[1].percent = 0; // A total of 50% is now allocated
+
+    await mockSplitsStore.mock.splitsOf
+      .withArgs(PROJECT_ID, timestamp, RESERVED_SPLITS_GROUP)
+      .returns(splits);
+
+    await mockTokenStore.mock.mintFor
+      .withArgs(
+        splitsBeneficiariesAddresses[0],
+        PROJECT_ID,
+        Math.floor(RESERVED_AMOUNT / splitsBeneficiariesAddresses.length),
+        true,
+      )
+      .returns();
+
+    await mockTokenStore.mock.mintFor
+      .withArgs(splitsBeneficiariesAddresses[1], PROJECT_ID, 0, true)
+      .returns();
+
+    await mockTokenStore.mock.mintFor
+      .withArgs(
+        projectOwner.address,
+        PROJECT_ID,
+        Math.floor(RESERVED_AMOUNT / splitsBeneficiariesAddresses.length),
+        false,
+      )
+      .returns();
+
+    expect(
+      await jbController.connect(caller).callStatic.distributeReservedTokensOf(PROJECT_ID, MEMO),
+    ).to.equal(RESERVED_AMOUNT);
+
+    const tx = await jbController.connect(caller).distributeReservedTokensOf(PROJECT_ID, MEMO);
+
+    // Expect one event per non-null split + one event for the whole transaction
+    await Promise.all([
+      await expect(tx)
+        .to.emit(jbController, 'DistributeToReservedTokenSplit')
+        .withArgs(
+          timestamp,
+          1,
+          PROJECT_ID,
+          [
+            splits[0].preferClaimed,
+            splits[0].percent,
+            splits[0].lockedUntil,
+            splits[0].beneficiary,
+            splits[0].allocator,
+            splits[0].projectId,
+          ],
+          RESERVED_AMOUNT / splitsBeneficiariesAddresses.length,
+          caller.address,
```

Comment:
ditto

Reply:
ok, i'll stop leaving these :) maybe do a scan of the PR to add these where they make sense.